types/mongodb/test/collection/insertX.ts

```diff
@@ -45,21 +78,154 @@ async function run() {
+  indexTypeResult1.insertedId;            // $ExpectType ObjectId
+  // should not remove types of existing fields
+  indexTypeResult1.ops[0].stringField;    // $ExpectType string
+  indexTypeResult1.ops[0].numberField;    // $ExpectType number
+  // should assign "any" type to any other field
+  indexTypeResult1.ops[0].randomField;    // $ExpectType any
+  // should do the same for insertMany
+  indexTypeResultMany1.ops[0]._id;          // $ExpectType ObjectId
+  indexTypeResultMany1.insertedIds;         // $ExpectType { [key: number]: ObjectId; }
+  indexTypeResultMany1.ops[0].numberField;  // $ExpectType number
+  indexTypeResultMany1.ops[0].stringField;  // $ExpectType string
+  indexTypeResultMany1.ops[0].randomField;  // $ExpectType any
+
+  /**
+   * test indexed types with custom _id (not ObjectId)
+   */
+  interface IndexTypeTestModelWithId {
+    _id: number;
+    stringField: string;
+    numberField?: number;
+    [key: string]: any;
+  }
+  const indexTypeCollection2 = db.collection<IndexTypeTestModelWithId>('testCollection');
+
+  const indexTypeResult2 = await indexTypeCollection2.insertOne({
+    _id: 1,
+    stringField: 'hola',
+    numberField: 23,
+    randomField: [34, 54, 32],
+    randomFiel2: 32,
+  });
+  const indexTypeResultMany2 = await indexTypeCollection2.insertMany([
+    { _id: 1, stringField: 'hola', numberField: 0 },
+    { _id: 2, stringField: 'hola', randomField: [34, 54, 32] },
+  ]);
+
+  // should only accept _id type provided in Schema
+  // $ExpectError
+  await indexTypeCollection2.insertOne({ _id: '12', stringField: 'hola', numberField: 23, randomField: [34, 54, 32], randomFiel2: 32 });
+  // $ExpectError
+  await indexTypeCollection2.insertMany([ { _id: '1', stringField: 'hola', numberField: 0 }, { _id: 2, stringField: 'hola', randomField: [34, 54, 32] } ]);
+
+  // should demand _id if it is defined and is not ObjectId
+  // $ExpectError
+  await indexTypeCollection2.insertOne({ stringField: 'hola', numberField: 23, randomField: [34, 54, 32], randomFiel2: 32 });
+  // $ExpectError
+  await indexTypeCollection2.insertMany([ { stringField: 'hola', numberField: 0 }, { _id: 12, stringField: 'hola', randomField: [34, 54, 32] } ]);
+
+  indexTypeResult2.ops[0]._id;       // $ExpectType number
+  indexTypeResult2.insertedId;       // $ExpectType number
+  indexTypeResultMany2.ops[0]._id;   // $ExpectType number
+  indexTypeResultMany2.insertedIds;  // $ExpectType { [key: number]: number; }
+
+  /**
+   * test indexed types with custom _id (ObjectId)
+   */
+  interface IndexTypeTestModelWithObjectId {
+    _id: ObjectId;
+    stringField: string;
+    numberField?: number;
+    [key: string]: any;
+  }
+  const indexTypeCollection3 = db.collection<IndexTypeTestModelWithObjectId>('testCollection');
+
+  // TODO: should not demand _id if it is ObjectId
+  // await indexTypeCollection3.insertOne({ stringField: 'hola', numberField: 23, randomField: [34, 54, 32], randomFiel2: 32 });
```

Comment:
Do you need to add these tests or they will be added at a later time?

Reply:
This is a valid use case but as I said above I couldn't add support for it with the current TS features